src/SmartTransactionsController.ts

```diff
@@ -796,13 +864,13 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       console.error('provider error', error);
     }
 
-    const requiresNonce = !txParams.nonce;
+    const requiresNonce = !txParams?.nonce;
     let nonce;
     let nonceLock;
     let nonceDetails = {};
 
     if (requiresNonce) {
-      nonceLock = await this.getNonceLock(txParams?.from);
+      nonceLock = await this.#getNonceLock(txParams?.from as string);
```

Comment:
`this.#getNonceLock` appears to throw an error if an `address` parameter is not supplied, meaning `txParams` being undefined results in `nonce{,Lock,Details}` all being undefined as well. Therefore, the following change should be equivalent to the original code while resolving the type issues.

```diff
@@ -864,18 +864,16 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
       console.error('provider error', error);
     }
 
-    const requiresNonce = !txParams?.nonce;
+    const requiresNonce = txParams && !txParams.nonce;
     let nonce;
     let nonceLock;
     let nonceDetails = {};
 
     if (requiresNonce) {
-      nonceLock = await this.#getNonceLock(txParams?.from as string);
+      nonceLock = await this.#getNonceLock(txParams.from);
       nonce = hexlify(nonceLock.nextNonce);
       nonceDetails = nonceLock.nonceDetails;
-      if (txParams) {
-        txParams.nonce ??= nonce;
-      }
+      txParams.nonce ??= nonce;
     }
     const submitTransactionResponse = {
       ...data,
```